### DISCUSSION TITLE
some stuff i get confused about in next js

### DISCUSSION THREAD

**Discussion by HamzaAmar** [vercel/next.js#18822] [Help] [answered]
hello all i have some question i dont understand in next js please can someone clear this stuff for me and thanks

# api route

1) is it obligatory to deploy api routes to lambdas functions ?
1) What is the benefits of deploying my api route to lambdas function ? 
1) it's not a problem if i work with api routes in big project rather than express app ?

# Data Fetching
4) Do we need getServerSideProps if yes can you give me some use case please?
2) can i have the same use case that getServerSideProps gives me with getStaticProps , Incremental Static Regeneration and Client side Data Fetching with swr?
2) its not a problem if i read and write to database with getStaticProps not with a cms ?

**Comment by mimoid-prog** [comment]
2. [https://github.com/vercel/next.js/discussions/17679](https://github.com/vercel/next.js/discussions/17679l)

**Reply by HamzaAmar** [reply]
Thank you so much from the reply but it's only answer for the last question . can some one have a response for the others and thanks

**Reply by HamzaAmar** [reply]
@jamesmosier  @Timer @timneutkens   any help please and am so sorry if i bother you guys

**Comment by jamesmosier** [comment, marked as answer]
> 1. is it obligatory to deploy api routes to lambdas functions ?

No. You can deploy Next.js apps with API routes anywhere.

> 2. What is the benefits of deploying my api route to lambdas function ?

There are many benefits and in my opinion the most important are they are cost effective to run and also allow you to run your app at the edge which means users will be served the content from the closest data center to their location.

> 3. it's not a problem if i work with api routes in big project rather than express app ?

It is up to you! I prefer to use Next.js API routes in order to keep all of my code within my Next app.

> 4. Do we need getServerSideProps if yes can you give me some use case please?

No, you do not need it unless you have a requirement to have some code run on the server before the page is rendered.

5. can i have the same use case that getServerSideProps gives me with getStaticProps , Incremental Static Regeneration and Client side Data Fetching with swr?

> I'm not certain I understand your question, but I'd recommend reading over [the docs](https://nextjs.org/docs/basic-features/data-fetching). I don't know your use case so I can't answer what would be best.

6. its not a problem if i read and write to database with getStaticProps not with a cms ?

It is not a problem. You can query data from a database in `getStaticProps` or `getServerSideProps` or even `getStaticPaths`.
